compiler/rustc_trait_selection/src/error_reporting/infer/suggest.rs

```diff
@@ -167,6 +167,14 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             exp_span, exp_found.expected, exp_found.found,
         );
 
+        let is_async_block = matches!(
```

Comment:
Please exhaustively match on `CoroutineKind`. Also, turning this into a match lets you just write `=> return,` rather than first storing this into bool then having to branch using an if-statement.

I believe this is leaving out `AsyncGen` so we suppress the this for `async gen {}` blocks.